composeApp/src/commonMain/kotlin/org/ooni/probe/App.kt

```diff
@@ -98,8 +98,10 @@ fun App(
     LaunchedEffect(Unit) {
         dependencies.bootstrapTestDescriptors()
         dependencies.bootstrapPreferences()
-        dependencies.configureDescriptorAutoUpdate()
-        dependencies.startDescriptorsUpdate(null)
+        if (dependencies.hasTestDescriptorInstalled()) {
+            dependencies.configureDescriptorAutoUpdate()
+            dependencies.startDescriptorsUpdate(null)
+        }
```

Comment:
This approach works for the app start update. But I feel this isn't the best approach for scheduling. If a user installs a link, but then deletes it at a later stage, it will continue with the daily background updates. I think it's better to match `observeAndConfigureAutoRun()`: continuously observe if there are any links installed, and cancel or schedule the background tasks.

Maybe the `startDescriptorsUpdate` call can also be done inside the observe domain class?

Reply:
Ah i see. Let me do that.